pkg/controllers/customroute/custom_route_controller.go

```diff
@@ -0,0 +1,132 @@
+			Namespace: destSecretNamespace,
+			Name:      destSecretName,
+		},
+		componentRoute: types.NamespacedName{
+			Namespace: componentRouteNamespace,
+			Name:      componentRouteName,
+		},
+		routeLister:    routeInformer.Lister(),
+		ingressLister:  ingressInformer.Lister(),
+		ingressSetter:  ingressSetter,
+		secretLister:   secretInformer.Lister(),
+		resourceSyncer: resourceSyncer,
+	}
+	return factory.New().
+		WithInformers(
+			ingressInformer.Informer(),
+			secretInformer.Informer(),
+		).
+		WithSyncDegradedOnError(operatorClient).
+		WithSync(controller.sync).
+		ResyncEvery(1*time.Minute).
+		ToController("componentRouteSecretSyncController", eventRecorder)
+}
+
+func (c *customRouteController) sync(ctx context.Context, syncCtx factory.SyncContext) error {
+	ingressConfig, err := c.ingressLister.Get("cluster")
+	if err != nil {
+		return err
+	}
+
+	// Update ingressConfig
+	route, err := c.routeLister.Routes("openshift-authentication").Get("oauth-openshift")
+	if err != nil {
+		return err
+	}
+
+	ingressConfigCopy := ingressConfig.DeepCopy()
+	c.updateIngressConfigStatus(ctx, configv1.Hostname(route.Spec.Host), ingressConfigCopy)
+
+	// Copy source secret to destination
+	destination := resourcesynccontroller.ResourceLocation{Namespace: c.destSecret.Namespace, Name: c.destSecret.Name}
+	source := c.componentRouteSecretSource(ingressConfigCopy)
+
+	return c.resourceSyncer.SyncSecret(destination, source)
+}
+
+func (c *customRouteController) componentRouteSecretSource(ingress *configv1.Ingress) resourcesynccontroller.ResourceLocation {
+	// check if expected componentRoute exists
+	if secretName := common.GetCustomRouteServingCertKeySecret(ingress, c.componentRoute.Namespace, c.componentRoute.Name); secretName != "" {
+		if _, err := c.secretLister.Secrets("openshift-config").Get(secretName); err == nil {
+			return resourcesynccontroller.ResourceLocation{Namespace: "openshift-config", Name: secretName}
+		}
+	}
+	return resourcesynccontroller.ResourceLocation{}
+}
+
+func (c *customRouteController) updateIngressConfigStatus(ctx context.Context, currentHostname configv1.Hostname, ingress *configv1.Ingress) error {
+	desiredComponentRoute := configv1.ComponentRouteStatus{
+		Namespace:        c.componentRoute.Namespace,
+		Name:             c.componentRoute.Name,
+		DefaultHostname:  configv1.Hostname("oauth-openshift." + ingress.Spec.Domain),
+		CurrentHostnames: []configv1.Hostname{currentHostname},
+		RelatedObjects: []configv1.ObjectReference{
+			{
+				Group:     routev1.GroupName,
+				Resource:  "routes",
+				Name:      "oauth-openshift",
+				Namespace: "openshift-authentication"},
+		},
+	}
+
+	if existingComponentRoute := common.GetComponentRouteStatus(ingress, c.componentRoute.Namespace, c.componentRoute.Name); existingComponentRoute == nil {
+		ingress.Status.ComponentRoutes = append(ingress.Status.ComponentRoutes, desiredComponentRoute)
+	} else {
+		*existingComponentRoute = desiredComponentRoute
```

Comment:
I prefer addres assignment to copying:
`existingComponentRoute = &desiredComponentRoute`